extensions/http-curl/processors/InvokeHTTP.cpp

```diff
@@ -443,6 +447,7 @@ void InvokeHTTP::route(const std::shared_ptr<core::FlowFile> &request, const std
   } else {
     if (request != nullptr) {
       if (penalize_no_retry_) {
+        logger_->log_debug("Flowfile has been penalized");
         session->penalize(request);
       }
```

Comment:
ProcessSession::penalize already produces a log message, so I think this is unnecessary.

Reply:
This was added to check whether the flowfile is being penalized in case of no retry relationship. Maybe the wording of the log message could be improved as it is used to test for PenalizeOnNoRetry property?

Reply:
You could check for that log message instead. This is how it's produced: (ProcessSession.cpp:202)
```
  logging::LOG_INFO(logger_) << "Penalizing " << flow->getUUIDStr() << " for " << penalization_period.count() << "ms at " << process_context_->getProcessorNode()->getName();
```

Try using `LogTestController::matchesRegex` to match this message. Something like this (untested) regex might work: `"Penalizing ([0-9a-f]+-)+[0-9a-f]+ for [0-9]+ms at invokehttp"`